include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -553,21 +607,7 @@ namespace VectorTools
     VectorType &                                               vec,
     const ComponentMask &                                      component_mask)
   {
-    Assert(dof_handler.get_fe().n_components() == function.n_components,
```

Comment:
Do you actually need to introduce your new two `internal::interpolate(...)` functions? Both `DoFHandler` and `hp::DoFHandler` have the member function `get_fe_collection()`.

I guess if you just replace `dof_handler.get_fe()` with `dof_handler.get_fe_collection()` in this assertion, you should be fine!